carbonmark-api/src/routes/retirements/[id]/provenance/get.ts

```diff
@@ -15,19 +15,28 @@ const handler = () =>
   ) {
     const sdk = gql_sdk(request.query.network);
 
-    // Finds the retirement using polygon-bridged-carbon. We can bypass this when we will have the trasaction hash in the Retire model of polygon-digital-carbon
-    const retirementRecord = (
+    const retirementRecord =
       await sdk.digital_carbon.getProvenanceRecordsByHash({
         hash: request.params.id,
-      })
-    ).provenanceRecords.at(0);
+      });
+    if (retirementRecord.retires.length === 0 || !retirementRecord.retires[0]) {
+      return reply.notFound();
+    }
 
-    if (retirementRecord == null) {
+    if (!retirementRecord.retires[0].provenance) {
       return reply.notFound();
     }
+    const registry =
+      retirementRecord.retires[0].credit.project.id.split("-")[0];
```

Comment:
Can you use the CreditId utility here?
https://github.com/KlimaDAO/klimadao/blob/staging/carbonmark-api/src/utils/CreditId.ts